src/idl_gen_cpp.cpp

```diff
@@ -883,11 +885,10 @@ class CppGenerator : public BaseGenerator {
   }
 
   std::string GetUnionElement(const EnumVal &ev, bool wrap_namespace,
-                              bool actual_type, bool native_type,
-                              bool wrap_native, const IDLOptions &opts) {
+                              bool native_type, const IDLOptions &opts) {
```

Comment:
Is it really needed to pass `opts`? I think we can use `opts_` member inside this method.

I don't know why `NativeName` requires the `opts` argument. As I can see it is called only with `opts_` member.
Might be @aardappel know a real reason.

Reply:
I was just asking myself that. I don't see a reason why `NativeName` would need to be `static`.

Reply:
It's actually called with parser_.opts from L.2846. Not sure if that's intentional.

Reply:
Ok. Let it be as it is with `opts` argument. This could be improved later.

Reply:
It is possible that is not intentional. Feel free to improve.